aws/data_source_aws_route53_resolver_rules_test.go

```diff
@@ -0,0 +1,103 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+func TestAccDataSourceAwsRoute53ResolverRules_basic(t *testing.T) {
+	dsResourceName := "data.aws_route53_resolver_rules.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t); testAccPreCheckAWSRoute53Resolver(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceAwsRoute53ResolverRules_basic,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(dsResourceName, "resolver_rule_ids.#", "1"),
+					resource.TestCheckResourceAttr(dsResourceName, "resolver_rule_ids.1743502667", "rslvr-autodefined-rr-internet-resolver"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDataSourceAwsRoute53ResolverRules_ResolverEndpointId(t *testing.T) {
+	rName1 := fmt.Sprintf("tf-testacc-r53-resolver-%s", acctest.RandStringFromCharSet(8, acctest.CharSetAlphaNum))
+	rName2 := fmt.Sprintf("tf-testacc-r53-resolver-%s", acctest.RandStringFromCharSet(8, acctest.CharSetAlphaNum))
+	ds1ResourceName := "data.aws_route53_resolver_rules.by_resolver_endpoint_id"
+	ds2ResourceName := "data.aws_route53_resolver_rules.by_rule_type"
+	ds3ResourceName := "data.aws_route53_resolver_rules.by_share_status"
+	ds4ResourceName := "data.aws_route53_resolver_rules.by_invalid_owner_id"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t); testAccPreCheckAWSRoute53Resolver(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceAwsRoute53ResolverRules_resolverEndpointId(rName1, rName2),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(ds1ResourceName, "resolver_rule_ids.#", "1"),
+					resource.TestCheckResourceAttr(ds2ResourceName, "resolver_rule_ids.#", "1"),
+					resource.TestCheckResourceAttr(ds3ResourceName, "resolver_rule_ids.#", "2"),
```

Comment:
Shared test cofiguration across data sources may be causing an issue when running all test in parallel. 

```
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -parallel 20 -run=TestAccDataSourceAwsRoute53ResolverRule -timeout 120m
?       github.com/terraform-providers/terraform-provider-aws   [no test files]
=== RUN   TestAccDataSourceAwsRoute53ResolverRule_basic
=== PAUSE TestAccDataSourceAwsRoute53ResolverRule_basic
=== RUN   TestAccDataSourceAwsRoute53ResolverRule_ResolverEndpointIdWithTags
=== PAUSE TestAccDataSourceAwsRoute53ResolverRule_ResolverEndpointIdWithTags
=== RUN   TestAccDataSourceAwsRoute53ResolverRules_basic
=== PAUSE TestAccDataSourceAwsRoute53ResolverRules_basic
=== RUN   TestAccDataSourceAwsRoute53ResolverRules_ResolverEndpointId
=== PAUSE TestAccDataSourceAwsRoute53ResolverRules_ResolverEndpointId
=== CONT  TestAccDataSourceAwsRoute53ResolverRule_basic
=== CONT  TestAccDataSourceAwsRoute53ResolverRules_ResolverEndpointId
=== CONT  TestAccDataSourceAwsRoute53ResolverRules_basic
=== CONT  TestAccDataSourceAwsRoute53ResolverRule_ResolverEndpointIdWithTags
--- PASS: TestAccDataSourceAwsRoute53ResolverRules_basic (21.14s)
--- PASS: TestAccDataSourceAwsRoute53ResolverRule_basic (45.18s)
--- FAIL: TestAccDataSourceAwsRoute53ResolverRules_ResolverEndpointId (244.39s)
    testing.go:568: Step 0 error: Check failed: Check 3/4 error: data.aws_route53_resolver_rules.by_share_status: Attribute 'resolver_rule_ids.#' expected "2", got "3"
--- PASS: TestAccDataSourceAwsRoute53ResolverRule_ResolverEndpointIdWithTags (256.00s)
FAIL
FAIL    github.com/terraform-providers/terraform-provider-aws/aws       256.053s
GNUmakefile:20: recipe for target 'testacc' failed
make: *** [testacc] Error 1
```

Reply:
Yes and also the existing state of any Route 53 Resolver Rules in the account.
I'll change the tests to be isolated from each other and any existing rules.

Reply:
Done.